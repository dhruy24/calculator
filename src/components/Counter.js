import React, { useState } from 'react'

function Counter() {
    const [count, setCount] = useState(0)
    return (
        <>
            <div data-testid="counter">Count:- {count}</div>
            <button data-testid="inc" className='px-4 px-2 bg-green-500 mr-2 mt-2' onClick={()=>setCount(prev=>prev+1)}>+</button>
            <button data-testid="dec" className='px-4 px-2 bg-green-500 mr-2 mt-2' onClick={()=>setCount(prev=>prev-1)}>-</button>
            <button data-testid="reset" className='px-4 px-2 bg-blue-500 mr-2 mt-2' onClick={()=>setCount(0)}>Reset</button>
        </>
    )
}

export default Counter