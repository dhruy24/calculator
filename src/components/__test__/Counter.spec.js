//import {render, fireEvent, screen} from "@testing-library/jest-dom"
import { fireEvent, render, screen} from '@testing-library/react';

import Counter from "../Counter"

test('increament counter', ()=>{
    //rendering component
    render(<Counter/>)

    //getting the elements
    const counter = screen.getByTestId('counter');
    const incButton = screen.getByTestId('inc');
    // const decButton = screen.getById("dec");
    // const reset = screen.getById("reset");

    expect(counter).toHaveTextContent('Count:- 0');

    // Click the increment button
    fireEvent.click(incButton);

    // Check if the count updated
    expect(counter).toHaveTextContent('Count:- 1');


})