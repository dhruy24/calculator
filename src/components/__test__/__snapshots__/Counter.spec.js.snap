// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Screenshot should match the screenshot 1`] = `null`;
